src/transformer/descriptor/type/type.ts

```diff
@@ -62,5 +58,10 @@ export function GetType(node: ts.Node, scope: Scope): ts.Node {
     return GetType(node.type, scope);
   }
 
+  if (ts.isCallExpression(node)) {
+    const declaration: ts.Declaration = TypescriptHelper.GetDeclarationFromNode(node.expression);
+    return GetType(GetReturnNodeFromBody(declaration as ts.FunctionDeclaration), scope);
```

Comment:
You may want to add some test for when is not a function declaration .


``` class Test {
a = test()
}

const test = () => {
  return 'ss'
}```

Reply:
no casting anymore, the change is quite big actually